core/src/main/resources/hudson/model/Computer/index.jelly

```diff
@@ -59,7 +59,11 @@ THE SOFTWARE.
         </j:if>
       </h1>
 
-      <j:if test="${it.offlineCause!=null and it.offline and !it.connecting}">
+      <j:if test="${it.temporarilyOfflineCause!=null}">
+        <st:include it="${it.temporarilyOfflineCause}" page="cause.jelly" />
+      </j:if>
+
+      <j:if test="${it.offlineCause!=null and it.offlineCause != it.temporarilyOfflineCause and it.offline and !it.connecting}">
```

Comment:
Better to handle it is a single Java method to enhance performance

Reply:
not sure if this is really a problem, agent overview pages are probably not visited that often.